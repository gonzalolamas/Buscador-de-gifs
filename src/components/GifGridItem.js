import React from "react";

export const GifGridItem = ({ title, url }) => {
  //console.log(id, title, url)
  return (
    <div className="card animate__animated animate__bounce animate__faster">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

/* 
Configurar

1. Enzyme
2. Enzyme to Json
3. debe de mostrar el componente correctamente
  * shallow
  * wrapper
  * wrapper .toMatchSnapshot()

Esto lo hacemos en setupTests, counterApp hicimos estas configuraciones
*/
