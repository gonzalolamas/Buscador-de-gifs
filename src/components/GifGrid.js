import React, { useState, useEffect } from "react";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url =
      "https://api.giphy.com/v1/gifs/search?limit=10&q=Rick&api_key=n1J096fYI33EKAhpV0O7SkGMHMzDoHCj";
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized.url,
      };
    });

    //console.log(gifs);
    setImages(gifs);
  };

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
