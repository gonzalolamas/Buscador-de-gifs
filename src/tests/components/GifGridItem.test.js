import React from 'react'
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'

import {GifGridItem} from '../../components/GifGridItem';

const title = 'un titulo'
const url = 'https://localhost/algo.jpg'

describe('Pruebas en GifGridItem', () => {
    test('debe de mostrar el componente correctamente', () => {

        const title = 'Hola, soy el t√≠tulo del componente'

        const url = 'https://imagenredireccion.svg'
        
        const wrapper = shallow(<GifGridItem title={title} url={url}/>)

        expect(wrapper).toMatchSnapshot()
    })
    
})