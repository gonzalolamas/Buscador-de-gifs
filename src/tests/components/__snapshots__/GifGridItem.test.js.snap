// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce animate__faster"
>
  <img
    alt="Hola, soy el título del componente"
    src="https://imagenredireccion.svg"
  />
  <p>
    Hola, soy el título del componente
  </p>
</div>
`;
